core/src/main/java/com/linecorp/armeria/common/sse/DefaultServerSentEvent.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.sse;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.Duration;
+import java.util.Objects;
+import java.util.function.Function;
+
+import javax.annotation.Nullable;
+
+import com.google.common.base.MoreObjects;
+
+/**
+ * A default implementation of the {@link ServerSentEvent} interface.
+ *
+ * @param <T> the type of the data
+ */
+final class DefaultServerSentEvent<T> implements ServerSentEvent<T> {
+
+    @Nullable
+    private final String id;
+    @Nullable
+    private final String event;
+    @Nullable
+    private final Duration retry;
+    @Nullable
+    private final String comment;
+    @Nullable
+    private final T data;
+
+    private final Function<T, String> dataStringifier;
```

Comment:
How about dropping this and changing the type of `data` to `String` for simplicity? A user can specify a converter anyway when using the factory methods in `ServerSentEvents`.